doc/man3/SSL_CTX_set_tlsext_servername_callback.pod

```diff
@@ -39,16 +41,27 @@ SSL_get_servername_type() returns the servername type or -1 if no servername
 is present. Currently the only supported type (defined in RFC3546) is
 B<TLSEXT_NAMETYPE_host_name>.
 
+SSL_set_tlsext_host_name() sets the server name indication ClientHello extension
+to contain the value B<name>. The type of server name indication extension is set
+to B<TLSEXT_NAMETYPE_host_name> (defined in RFC3546).
+
 =head1 NOTES
 
 Several callbacks are executed during ClientHello processing, including
 the early, ALPN, and servername callbacks.  The early callback is executed
 first, then the servername callback, followed by the ALPN callback.
 
+The SSL_set_tlsext_host_name() function should not be called from a server
+context, typically in which either the context is created by a explicit 'server'
+method or, a server side specific function is called (e.g. L<SSL_accept> or
+L<SSL_set_accept_state>). Otherwise, B<name> will be ignored as soon as the
+server starts to accept SSL/TLS handshakes.
```

Comment:
This wording could probably be tidied up.  How about

SSL_set_tlsext_host_name() should only be called on SSL objects that will act as clients; the configured name will be ignored for SSL objects later passed to L<SSL_accept> or L<SSL_set_accept_state>.